hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -310,6 +310,19 @@ protected synchronized boolean areSomeRegionReplicasColocated(Cluster c) {
     return false;
   }
 
+  private synchronized boolean areThereIdleRegions(Cluster c){
+    boolean isServerExistsWithMoreRegions=false;
+    boolean isServerExistsWithZeroRegions=false;
+    for (ServerName server:c.servers){
```

Comment:
This is iterating over many data structures and complex. Why cant we just iterate over the regionsPerServer data structure alone and populate the booleans as needed?

Reply:
thanks, that was an idea as well, but I wasn't sure about it, I'll fix it